code/game/gamemodes/traitor/traitor.dm

```diff
@@ -59,9 +59,11 @@
 		if (!possible_traitors.len)
 			break
 		var/datum/mind/traitor = pick(possible_traitors)
+		possible_traitors -= traitor
+		if(traitor.special_role)
+			continue
```

Comment:
Actually I think this will fix it.

traitorchan calls this pre_setup, which is missing the continue if someone already has a special role.